tests/Earthfile

```diff
@@ -1473,6 +1474,12 @@ test-cache-mode:
       --should_fail=false \
       --output_contains="drw-r--r-- /cache-folder"
 
+test-unsafe-cache:
+  DO +RUN_EARTHLY \
+      --earthfile=unsafe-cache.earth \
+      --target=+test \
+      --should_fail=false
+
```

Comment:
Can you also add a test where the cache is used across two different Earthfiles?

Reply:
Do we have some tests using multiple Earthfiles right now @alexcb ?

Reply:
The `import` test is perhaps an example: https://github.com/earthly/earthly/blob/main/tests/Earthfile#L22. When there are multiple files involved, we usually put them under their own dir under `test/`.

Reply:
I don't think `import` is a great example because it doesn't use the `DO +RUN_EARTHLY`, which requires the dev version of earthly to be used for running the tests rather than the earthly-in-earthly approach we tend to favour.

Instead I would suggest looking at:

`pass-args-test`, which copies two different files in:
https://github.com/earthly/earthly/blob/3a9567eb17b1d4f91b495b9b505500f3df067715/tests/Earthfile#L515-L518

or, `allow-privileged-import-test` is another example which performs an `echo` to produce a secondary Earthfile: 
https://github.com/earthly/earthly/blob/3a9567eb17b1d4f91b495b9b505500f3df067715/tests/Earthfile#L502C1-L509

That's not to say you can't combine these ideas together, and create a directory for your tests, here's a more complicated example that still uses earthly-in-earthly: https://github.com/earthly/earthly/blob/3a9567eb17b1d4f91b495b9b505500f3df067715/tests/locally-in-command/Earthfile#L12-L20